### ISSUE TITLE
App crashes when a penultimate card is swiped, and subsequently the last card is clicked

### ISSUE THREAD

**Issue by niranjan-nagaraju** [niranjan-nagaraju/notification-center#7]
App crashes when a penultimate card is swiped, and then subsequently when the next card's header is clicked to expand, the adapter crashes with an IndexOutOfBoundsException.
```
12-12 17:05:05.746  8987  8987 E AndroidRuntime: Process: com.vinithepooh.notifier, PID: 8987
12-12 17:05:05.746  8987  8987 E AndroidRuntime: java.lang.IndexOutOfBoundsException: Index: 18, Size: 18
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at java.util.ArrayList.get(ArrayList.java:437)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at com.vinithepooh.notifier.Ntfcns_adapter$1.onClick(Ntfcns_adapter.java:175)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.view.View.performClick(View.java:6669)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.view.View.performClickInternal(View.java:6638)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.view.View.access$3100(View.java:789)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:26145)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6863)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
12-12 17:05:05.746  8987  8987 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```
Other times, the crash is due to a NullPointerException

```
12-12 14:42:13.004  7236  7236 E AndroidRuntime: Process: com.vinithepooh.notifier, PID: 7236
12-12 14:42:13.004  7236  7236 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Notification android.service.notification.StatusBarNotification.getNotification()' on a null object reference
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at com.vinithepooh.notifier.Ntfcns_adapter$1.onClick(Ntfcns_adapter.java:174)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.view.View.performClick(View.java:6669)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.view.View.performClickInternal(View.java:6638)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.view.View.access$3100(View.java:789)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:26145)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6863)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:537)
12-12 14:42:13.004  7236  7236 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by niranjan-nagaraju** [comment]
**Issue root-cause:**
   The adapter positions for the items below the removed 'position' does not change and retains their old values despite 
`notifyItemRemoved(position);`

Adding a 
`notifyItemRangeChanged(position, dataset.size());`

for both remove and insertion at 'position' should fix the adapter positions for all cards below the remove/insert positions.

Reference: https://stackoverflow.com/questions/45413292/recyclerview-does-not-update-after-removing-an-item/45413953#45413953

**Comment by niranjan-nagaraju** [comment]
Crash fixed at commit: 9401719
Added `notifyItemRangeChanged()` for all indices below remove/insert position upon swipe/restore.
